include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -1277,7 +1279,8 @@ namespace VectorTools
 
       typedef MatrixFreeOperators::MassOperator<dim, fe_degree, n_q_points_1d, 1, Number> MatrixType;
       MatrixType mass_matrix;
-      mass_matrix.initialize(matrix_free);
+      mass_matrix.initialize(std_cxx11::make_shared<
+                             const MatrixFree<dim,typename VectorType::value_type> > (matrix_free));
```

Comment:
@Rombur i see now what you meant. If it's more complicated to do what @kronbichler suggest, I am ok to change the public interface of `project` to take a shared pointer. It would make like easy and consistent with now recommended way of storing `MatrixFree` data in a shared pointer.